runtime/hashtable.go

```diff
@@ -119,6 +119,9 @@ func (t *mixedTable) next(k Value) (next Value, v Value, ok bool) {
 		isInt = true
 	} else {
 		i, isInt = ToIntNoString(k)
+		if isInt && i == 0 {
+			return t.hashTable.next(IntValue(0))
+		}
```

Comment:
I think this is slightly better:

```suggestion
		if isInt && i <= 0 {
			isInt = false
		}
```